src/D2L.CodeStyle.TestAnalyzers/NUnit/CategoryAnalyzer.cs

```diff
@@ -221,24 +235,29 @@ ISymbol symbol
 
 		private static bool TryLoadNUnitTypes(
 			Compilation compilation,
-			out NUnitTypes types
+			[NotNullWhen( true )]
+			out NUnitTypes? types
 		) {
-			INamedTypeSymbol categoryAttribute = compilation.GetTypeByMetadataName( "NUnit.Framework.CategoryAttribute" );
+			INamedTypeSymbol? categoryAttribute = compilation.GetTypeByMetadataName( "NUnit.Framework.CategoryAttribute" );
 			if( categoryAttribute == null || categoryAttribute.TypeKind == TypeKind.Error ) {
 				types = null;
 				return false;
 			}
 
-			ImmutableHashSet<INamedTypeSymbol> testAttributes = ImmutableHashSet
-				.Create<INamedTypeSymbol>(
-					SymbolEqualityComparer.Default,
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TestAttribute" ),
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseAttribute" ),
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseSourceAttribute" ),
-					compilation.GetTypeByMetadataName( "NUnit.Framework.TheoryAttribute" )
-				);
+			ImmutableHashSet<INamedTypeSymbol> testAttributes = new[] {
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TestAttribute" ),
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseAttribute" ),
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TestCaseSourceAttribute" ),
+				compilation.GetTypeByMetadataName( "NUnit.Framework.TheoryAttribute" )
+			}.Where( x => x != null )!
+			.ToImmutableHashSet<INamedTypeSymbol>( SymbolEqualityComparer.Default );
 
-			INamedTypeSymbol testFixtureAttribute = compilation.GetTypeByMetadataName( "NUnit.Framework.TestFixtureAttribute" );
+			INamedTypeSymbol? testFixtureAttribute = compilation.GetTypeByMetadataName( "NUnit.Framework.TestFixtureAttribute" );
+
+			if( testFixtureAttribute == null ) {
+				types = null;
+				return false;
+			}

```

Comment:
(reasonable imo) behaviour change

Reply:
(clearly its not crashing today, so maybe in practice not, but I suppose NUnit could rename / move these types in the future.)